chore(release): 3.19.0

sfoster1 opened the pull request:
This is the release pr for 3.19.0.

Kadee80 committed:
feat(protocol-designer): Default tc state for to previous saved state (#5668)

closes #5577

Kadee80 committed:
fix(protocol-designer): Default to Gen1 for TC in edit modules modal (#5672)

closes #5669

mcous committed:
ci: use Python 3.7 in CI (#5679)

b-cooper committed:
fix(app): cal check: tweak copy, fix weird video asset, wrap tip rack names (#5670)

* remove extremely
* replace video asset with mysterious vertical bars
* title bar button should be exit instead of back
* tipracks on deck should have name ui wrap

Laura-Danielle committed:
refactor(api): Add in an index file to keep track of labware calibrations (#5691)

shlokamin committed:
feat(protocol-designer): add thermocycler command creator (#5683)

* WIP - adding getThermocyclerStaterDiff util

* add TC module state to fixture and rename variable

* feat(protocol-designer): add thermocycler command creator

#5600

* hook up thermocyclerFormToArgs

* reorder imports

Commit:
feat(api,robot-server): Add API hardware simulation from file (#5680)

* basic session lifecycle integration tests

* POST /sessions should return 201

* just cleaning up the docstrings

* add simulator_setup module for creating a simulator from a json file.

* fix loading of modules.

* define a simulator configuration for dev.

* lint error

* comments

* add --custom-hardware-simulator-file to simulate app. allows loading a simulator from a json file.

* remove humiliating print

* rename simulators/dev.json to test.json. create a new test.env file for running integration tests.

* more tests.

* gitignore json files in simulators folder. also add a readme about the file format

* lint

sfoster1 committed:
fix(api): respect port in tools arghandler (#5700)

The central args handler for our tools scripts didn't actually pass its
port argument when building its hardware controller which breaks a lot
of smoothie connections on non-linux.

IanLondon committed:
refactor(protocol-designer): remove misc unused files/exports (#5686)

Closes #5488

IanLondon committed:
refactor(protocol-designer): fix type annotation on engageMagnet (#5687)

IanLondon committed:
feat(protocol-designer): implement empty getDisabledFields for tc step (#5692)

Commit:
refactor(robot-server): remove item router and model

the item endpoints served us well as examples of v2 api, but we've graduated and they can be retired. (#5701)

Commit:
feat(api): cache_instruments when creating simulator.

b-cooper committed:
refactor(app): incorporate fast api session into cal check (#5690)

Deprecate the old calibration check domain layer that relied on the aiohttp api. Connect calibration check components to the new sessions domain layer.

Co-authored-by: amit lissack <amit@opentrons.com>

IanLondon committed:
feat(protocol-designer): add substeps for TC State step (#5695)

Closes #5694

mcous committed:
chore(release): merge changes from 3.18.0 into edge

sfoster1 committed:
chore(api): add util to plot session state machines (#5713)

This will parse the calibration check session (and eventually later
sessions) and produce a nice pdf graph if you have graphviz installed.
This can be useful for, for instance, noticing if there's a state that's
never used because it has no edge (hint hint).

To use, update your pipenv and install graphviz from somewhere (on
linux, your package manager; on OSX, homebrew; on windows, who knows)
and make sure its `dot` utility is on your path.

IanLondon committed:
feat(protocol-designer): implement ModuleTag for TC state (#5697)

Closes #5572

IanLondon committed:
feat(protocol-designer): add lid cooling hint (#5704)

Closes #5574

sfoster1 committed:
chore(api): plot_session: add conditions to plot (#5719)

Commit:
fix(robot-server): session commands with empty data field should be accepted

Laura-Danielle committed:
refactor(api): Modify thresholds, take into account safety moves (#5676)

sfoster1 committed:
feat(app): display the serials of attached pipettes (#5711)

Displays serial numbers in the pipette info page.

Closes #4370

Commit:
refactor(app,robot-server): Slight changes to sessions api (#5684)

* makes some naming more consistent

* session details are no longer returned in the metadata object.

* remove code from Error schema. it's redundand.
change /sessions/:sid/commands to /sessions/:sid/commands/execute as it actually creates and executes a a command.

* add epic to fetch session on successful command response

* fix failing test

* some slight typing changes to match server schema.

b-cooper committed:
fix(app): update cal height check video assets (#5723)

shlokamin committed:
feat(protocol-designer): add timeline error when aspirating/dispensing from TC with lid closed (#5705)

* feat(protocol-designer): add timeline error when aspirating from TC with lid closed

add timeline error when aspirating from TC with lid closed

close #5573

* import helper from index.js

* have util only take params from robot state that it needs

* return error when dispensing into TC with pipette collision

* import constant from shared-data

* fix spelling mistake

* update error copy

* add test case

Commit:
test(robot-server): integration tests for cal check and deck cal (#5718)

* skeleton of calibration check test flow.

* check that the currentStep is where it should be after each command.

* use yaml anchors and aliases

* check presence of comparisons object sub-fields by step

* skeleton of deck calibration test

* fill out deck calibration tests

* fix rebase related issues

* add /execute to commands path

Laura-Danielle committed:
refactor(api): Check tiprack thresholds after tip pickup (#5708)

Laura-Danielle committed:
fix(api): Fix determine thresholds function (#5729)

sanni-t committed:
feat(app, app-shell): add config migration (#5733)

Co-authored-by: Mike Cousins <mike@cousins.io>

sfoster1 committed:
fix(api): proper hardware controller teardown (#5739)

opentrons_execute was printing an exception when it stopped, because we
weren't properly tearing down the module watcher and were relying on the
interpreter to do so - not great. We now explicitly tear down the
hardware control stack.

When we tear down the hardware controller, we cancel some running tasks.
When you cancel a running task, it pends a CancelledException, but the
task needs to execute on the next loop spin, so we need to spin the
loop. This is the complex annoying logic in
thread_manager.__getattribute__.

Closes #5728

mcous committed:
chore(release): merge changes from 3.18.1 into edge

mcous committed:
chore(js): specify exact versions for production dependencies (#5730)

Closes #5067

Commit:
feat(api): use lru_cache on get_all_adv_settings (#5731)

theosanderson committed:
feat(api): add multiplication operator(s) to types.Point (#5742)

This allows scaling an offset by a fixed amount, making it easy to combine e.g. a direction vector and a distance.

Usage:
Point(1,2,3)*2 => Point(2, 4, 6)

b-cooper committed:
refactor(app): add results summary to cal check complete (#5732)

Differentiate between deck transform and instrument offset problems on front end of cal check flow. Add results raw data and summary to complete check page for user acceptance testing.

b-cooper committed:
chore(release): v3.19.0-alpha.0

sfoster1 committed:
fix(api): catch factory tests up to gpio changes (#5747)

Fix some circular imports in factory tests and change them to use the
right gpio interface for setting button lights

b-cooper committed:
Merge branch 'release_3.19.0-alpha.0' into edge

Commit:
fix(robot-server): make sure fastapi validation errors are properly documented (#5748)

Kadee80 committed:
feat(protocol-designer): Add ending hold state fields to TC profile (#5750)

closes #5519

Commit:
refactor(api,robot-server): move calibration check out of server (#5740)

* move calibration check modules to calibration package

* break apart calibration check modules

dozercodes committed:
test(robot-server): Add integration tests for robot endpoints (#5752)

IanLondon committed:
feat(protocol-designer): create barebones setup for TC profile steps (#5751)

sfoster1 committed:
chore: refactor python makefiles (#5760)

Pull out common code from python makefiles because it's getting a bit
ridiculous.

python definitions and useful macros are in scripts/python.mk; push
stuff in push.mk

Kadee80 committed:
fix(protocol-designer): Add scroll to top on export modal display (#5744)

closes #5422

dozercodes committed:
test(robot-server): Add tests for pipettes and motors endpoints (#5761)

* test(robot-server) add simple get test for pipettes

* Add simple tests for motors endpoint

shlokamin committed:
feat(components): add useConditionalConfirm hook to shared components (#5735)

* add test coverage around useConditionalConfirm hook

* rename variables

* move conditionalConfirm hook to shared components

* update text in test wrapper to be more clear

* update test names

* use const instead of let

* rename requireConfirmation to showConfirmation

* rename test file to use lowercase

* only return one confirm callback, rename cancelConfirm to cancel

* add JSDoc comments

sfoster1 committed:
fix: fix makefile refactor for push tasks (#5773)

This didn't quite work for a couple reasons:
- needed to fix some things specifically in the update-server package
- push-systemd-unit needs to remount the filesystem

sfoster1 committed:
refactor(shared-data): add python bindings skeleton (#5766)

This adds the skeleton of python bindings for shared-data actually
resident in shared-data, like the js bindings. This is the prequel to
moving direct shared data handling to these bindings and adding a lot of
type definitions.

Kadee80 committed:
feat(protocol-designer): Add profiles settings to TC profile form (#5771)

mcous committed:
feat(app): add network interface collection to system-info (#5764)

Closes #5397

SyntaxColoring committed:
docs(api): Various edits to the module docs (#5604)

Various restructurings and revisions for clarity and readability.

shlokamin committed:
fix(api): convert extra seconds to minutes in delay command (#5755)

* fix(api): convert extra seconds to minutes in delay command publisher

fix #5414

* filter seconds and minutes through timedelta, add test for delay command

* use pytest.mark.parametrize decorator

Kadee80 committed:
fix(protocol-designer): Remove errant p tag (#5786)

IanLondon committed:
feat(protocol-designer): disable save for dynamic field errors (#5769)

Part of #5737 - only addresses field-level, not form-level

b-cooper committed:
fix(app): fix up calibration check copy (#5779)

IanLondon committed:
feat(protocol-designer): implement form-level errors for dynamic profile fields (#5784)

addresses parts of #5737

IanLondon committed:
feat(js): update lodash to 4.17.15 (#5788)

theosanderson committed:
refactor(api): raise GPIO warning, and add detail (#5782)

This raises the level of the warning about the failure to initialise a GPIO device from `info` to `warning`, making it more likely that it will be displayed (e.g. in Jupyter). Without this there is no indication given to the user of e.g. a Jupyter notebook of why their commands to control lights are doing nothing. It also adds some detail about what functionality is likely to be affected.

theosanderson committed:
feat(api): add info to debug warning on how IS_ROBOT is determined (#5783)

If you run a robot without RUNNING_ON_PI in the environment you get this warning message (and substantially reduced functionality), but it is not clear how to resolve it. This provides that information. In my case it was because I was invoking one Python script from another with subprocess.

theosanderson committed:
fix(api): Enable audio on GPIO startup (#5787)

The GPIO pin that allows the Pi to output audio was automatically enabled prior to https://github.com/Opentrons/opentrons/pull/5381 . This is relied upon by e.g. this test https://github.com/Opentrons/opentrons/blob/751d9a402fa5d1f936df43ed12fd7387be23237f/api/src/opentrons/tools/factory_test.py#L217, and allows users to make their robots make sounds.

This should restore that functionality by enabling this pin during GPIO setup.

mcous committed:
refactor(js): start preparing JS codebase for Flow's types-first mode (#5785)

SyntaxColoring committed:
docs(api): Fix syntax errors in Thermocycler profile examples (#5792)

The examples forgot to put quotation marks around the dict keys.

mcous committed:
fix(app): clear <input> value after robot update file select (#5789)

Fixes #5781

mcous committed:
refactor(components): ensure primitives use consistent style props (#5778)

mcous committed:
style(js): prefer const for non-reassigned values (#5790)

Kadee80 committed:
feat(protocol-designer): Style dynamic step fields (#5795)

closes #5513

Commit:
refactor(robot-server): move rpc and v1 endpoints into a legacy package. (#5798)

move json_api out of models and under service as it is only part of v2 endpoints.
move session routers and models into service.session package.

sfoster1 committed:
refactor(shared-data): new steps per mm for gen2 multi 2.1 (#5797)



Co-authored-by: XiaoLing-git <40679757+XiaoLing-git@users.noreply.github.com>

Laura-Danielle committed:
fix(api): Fix return tip behavior in check session (#5772)

ahiuchingau committed:
feat(api, app): implement automatic door safety stop feature (#5706)

* feat(api, app): implement automatic door safety stop feature

closes #2820, closes #2752

Co-authored-by: Seth Foster <seth@opentrons.com>

b-cooper committed:
refactor(app): add analytics event to cal check session exit (#5780)

Commit:
feat(robot-server): Mock up of access control models and router. (#5749)

A sketch of what the access control API will look like when we are ready to implement it

closes #5746

b-cooper committed:
fix(app): interpolate session type into session exit event name (#5804)

mcous committed:
refactor(app): fully annotate types of legacy RPC API actions (#5799)

b-cooper committed:
feat(api): add skeleton state machine instance for tip length calibration (#5805)

Create an instance of the calibration state machine that defines the valid transitions, triggers,
and states that the tip length calibration flow can move through. Note: this does not yet add
callbacks or side effects to actually interact with the hardware controller or read/writing
calibration data.

Closes #5605

mcous committed:
refactor(app): explicitly annotate types of all exports in app (#5800)

IanLondon committed:
feat(protocol-designer): add support for TC cycles (#5809)

Addresses the behavioral parts of #5516, #5517, and #5518 but the styling parts of those tickets will require another PR

Kadee80 committed:
feat(protocol-designer): Add validation to profile settings and end hold (#5802)

closes #5520 closes #5753

sanni-t committed:
feat(app, app-shell): add ability to disable discovery cache (#5759)

Laura-Danielle committed:
refactor(api): Reduce pick up tip current for multichannel pipettes (#5812)

mcous committed:
refactor(labware-library): annote types of module exports (#5803)

shlokamin committed:
feat(test): add test id to Pipette Select component (#5791)

add test id to Pipette Select component for use in cypress test

#5724

mcous committed:
refactor(app): add type annotation missing due to merge timing (#5821)

mcous committed:
refactor(protocol-designer): annotate export types in several redux modules (#5817)

Kadee80 committed:
feat(protocol-designer): Add profile step numbers and tooltips (#5819)

closes #5515

IanLondon committed:
feat(protocol-designer): finish form-level errors for dynamic fields (#5818)

Closes #5814

* Show form-level errors for steps that are inside a cycle (not just top-level steps)
* Disable save when form-level errors are present

IanLondon committed:
refactor(protocol-designer): annotate more PD exports (#5824)

mcous committed:
refactor(protocol-designer): annotate types of connect'd components (#5827)

mcous committed:
refactor(protocol-designer): annotate exports of functional components (#5825)

sfoster1 committed:
refactor(api, shared-data): move py shared data accessors (#5777)

The JS stack has integrated type definitions and bindings for accessing the content in shared_data. Now the Python stack does too.

This adds a new python package, opentrons_shared_data, which will be built and uploaded to pypi with the same rules as the api package. It is loaded into the api and robot server pipenv spaces editably, and the api package marks it as a dependency to pypi.

opentrons_shared_data has little code - most of it is slightly-customized accessors for the json definitions - but it has a lot of types, types that define the structure of the shared data files and are ensured to be correct using the runtime type checker package typeguard in the unit tests (there is still no runtime type checking in the production code).

In addition, the api package and tests have been modified to use the types and loaders in opentrons_shared_data instead of directly accessing the content.
New packa
shared data now lives in opentrons_shared_data, with accessors identical
to opentrons.system.shared_data.

Also, add py.typed to the opentrons package so mypy can typecheck it externally.

Co-authored-by: Brian Arthur Cooper <brian@opentrons.com>

SyntaxColoring committed:
build(make): name targets "setup"/"teardown" instead of "install"/"uninstall" (#5823)

IanLondon committed:
refactor(protocol-designer): add annotations for more PD modules (#5833)

* Add annotations to load-file, localization, step-forms, and steplist
* snuck in a type def for i18next's i18n.t fn

sanni-t committed:
fix(api): prevent thermocycler wait_for_temp() from exiting prematurely (#5807)

Closes #5602

Kadee80 committed:
feat(protocol-designer): Add step grouping and styles for tc cycles (#5836)

closes #5516 closes #5517 closes #5518

sfoster1 committed:
fix(shared-data): recursively specify subpackages (#5844)

When you pass packages to setup.py you need to pass all the subpackages!
There's even a special helper just for that! I didn't use it! We didn't
get any subpackages and you can't tell from the tests because they use
editable installs! Oops!

theosanderson committed:
Removed many database migration print statements (#5847)

This removes the database migration statements that are printed every time the robot module is imported, which are confusing for users, in favour of `log.info` messages. It leaves print statements intact when migration actually happens so that if the interface does not lock up without any information on progress.

shlokamin committed:
refactor(components): add data id to pipette dropdown options (#5822)

* feat(test): add data id to pipette dropdown options

add data id to pipette dropdown options

#5724

* change variable naming to be more consistent

* use Flex styled component

* Update components/src/instrument/PipetteSelect.js

use relative import

Co-authored-by: Mike Cousins <mike@opentrons.com>

* omit non semantic div in cy.get

Co-authored-by: Mike Cousins <mike@opentrons.com>

mcous committed:
refactor(protocol-designer): add type exports to class components (#5834)

shlokamin committed:
feat(test): add button names to edit pipette, swap pipette, and continue to liquids (#5826)

add button names to edit pipette, swap pipette, and continue to liquids

#5724

Commit:
feat(robot-server): session manager (#5796)

* skeleton of session and session manager
* session manager with tests
* asyncify session and sessionmanager
* fix errors from rebase
* add command_execution package and start check_session implementation.
* Session moved to session_types and renamed BaseSession.
* calibration check session ported to new session manager. integration test works. unit tests need work.
* correct the calibration check creation and implement clean up
* CheckSession tests passing.
* rename CheckBaseSession to CheckSession
* unit tests pass.
* the most recent session is the active one. this may change in the future.
* add command execution errors for calcheck integration test.
* multi session integration test
* Replace HardwareAPILike with ThreadManager
* fix lint errors
* rename null session
* add some logging.
* replace IdentifierType alias with actual NewType
* rename created_on to created_at as it's a datetime
* Convert Command to several dataclasses. Create Command before passing on to Executor.


closes #5763
closes #5765

b-cooper committed:
fix(app): fix cal check tiprack images at large window sizes (#5841)

shlokamin committed:
refactor(protocol-designer): add names to edit, add, and delete module buttons in PD (#5832)

* feat(test): add names to edit, add, and delete module buttons in PD

add names to edit, add, and delete module buttons in PD

#5724

* remove assertions on slot position

* use lodash upperFirst to format button name

shlokamin committed:
feat(protocol-designer): add atomic thermocyclerRunProfile command creator (#5843)

* feat(protocol-designer): create atomic thermocyclerRunProfile command creator

create atomic thermocyclerRunProfile command creator

#5839

* consolodate TC command types

sfoster1 committed:
fix(robot-server): preserve newlines in text logs (#5850)

The fastapi automatic encoder for responses was doing a bit too much
when we returned a raw string in the logs response. By using the
Response class directly, we can specify the encoding as text/plain if
we're returning text logs instead of letting fastapi infer it.

Closes #5846

IanLondon committed:
feat(protocol-designer): implement formToArgs for TC profile (#5842)

Closes #5838

sfoster1 committed:
fix(build): allow test pypi uploads to fail again (#5849)

The shared-data pr had a small makefile refactor to coalesce the deploy
and deploy-staging python makefile tasks, but what that forgot is that
uploads to test pypi must be allowed to fail, since every edge build
in the same version will upload a wheel with the same name, which pypi
doesn't allow.

This adds that functionality back to the python_upload_package makefile
function

It also adds a bit of complexity to allow callers to specify a prefix
command since the shared-data task needs to execute in a subdirectory.

mcous committed:
refactor(js): enable Flow's types-first mode and fix errors (#5854)

Laura-Danielle committed:
refactor(api): Separate out pipette tolerances further in cal check (#5853)

sfoster1 committed:
chore(build): dev wheel versions on edge builds (#5855)

These get uploaded to test pypi (where they are by and large never
downloaded, but it allows us to dry-run uploads to production pypi).
Test pypi doesn't allow you to reuse filenames, just like real pypi, and
each edge build in the same version will have the same wheel filename.
In the past, we overcame this by allowing the test pypi upload to fail,
but then... that doesn't really test anything.

Luckily, PEP440[1] which defines python versioning conventions, allows a
.devN tag to be at the end of versions, with the intent being that the N
is a build number. So this pr adds the capability to specify a
BUILD_NUMBER= environment variable - which travis edge builds now do -
which will get added to
- the wheel filenames
- the actual versions encoded in the wheel metadata
- the required version for opentrons_shared_data in opentrons

and thus allow us to upload to pypi on every edge build.

b-cooper committed:
fix(api): return rail lights to original state post cal-check (#5861)

sfoster1 committed:
fix(api): do not allow v1 imports in v2 protocols (#5859)

If you import v1 protocols in your v2 protocol, it slows things down and
kinda messes them up since those modules are designed to do a lot of
work on import, and it's work that actively shouldn't be done unless
necessary.

If you then _use_ those modules, it can play absolute havoc with
assumptions inside the protocol session machinery.

This is just straight up not supported, and this pr adds an error that
will occur when you try and upload a protocol that does this.

Closes #5852

theosanderson committed:
refactor(api): lower the log level for warnings that an instrument is not attached to one mount (#5857)

It is reasonable to use the pipette with only one pipette attached but this results in many error level warnings, e.g.:

alarm/error outside hard halt: error:no R instrument found
alarm/error outside hard halt: error:no R instrument found
alarm/error outside hard halt: error:no R instrument found
alarm/error outside hard halt: error:no R instrument found

This lowers the priority of those warnings by moving their log-level to info. It still raises the SmoothieError exception.

Fixes #5770

Kadee80 committed:
feat(protocol-designer): Add number maskers to tc duration fields (#5848)

closes #5737

ahiuchingau committed:
feat(api, docs): add GPIO rail lights and door control to ProtocolContext (#5862)

* feat(api): add GPIO controls (rail lights & door) to protocol context

closes #5775 and closes #5703

Laura-Danielle committed:
chore(bump): v3.19.0-alpha.1 (#5866)

Commit:
fix(api): disallow calling Session._run while protocol is running. (#5831)

this attempts to fix a deadlock that can occur if a protocol has a pause step.
also if user clicks start run very quickly over and over.

closes #5239

IanLondon committed:
feat(protocol-designer): create barebones TC profile substeps (#5865)

sets up #5522 but without styling

IanLondon committed:
feat(protocol-designer): implement thermocyclerProfileStep CC (#5858)

Closes #5835

shlokamin committed:
feat(protocol-designer): add TC profile state updater (#5856)

add TC profile state updater

#5837

Kadee80 committed:
fix(protocol-designer): Correct tc profile fields sizing (#5870)

closes #5851

shlokamin committed:
test(protocol-designer): fix broken TC state test (#5872)

b-cooper committed:
fix(app): delete local records of session if id not found  (#5863)

In order to prevent stale/dangling sessions in redux state despite their non-existence on a given
robot, if we ever receive a 404 not found for a session we're attempting to fetch or delete, we will remove the local reference to that session and sessionId.

Kadee80 committed:
feat(protocol-designer): Reset TC form on form type toggle (#5871)

closes #5754

IanLondon committed:
feat(protocol-designer): style TC profile substeps (#5876)

Addresses more of #5522, lacking "Ending hold" section

sfoster1 committed:
chore(shared-data): add python/ makefile (#5875)

and put python stuff in there. that prevents an issue where the
wheelname variable would run at makefile rule build time and create
spurious venvs and pipenv files in shared-data/

theosanderson committed:
refactor(api): add set_acceleration to SimulatingDriver (#5879)

This adds the set_acceleration command to SimulatingDriver meaning that this command can be used in simulated protocols.

sfoster1 committed:
fix(scripts): make python_build_utils 2.7 compatible again (#5886)

there's even a big warning at the top! That I just missed. This needs to
be able to run under python 2.7 for annoying buildroot reasons.

IanLondon committed:
feat(protocol-designer): add ending hold to TC profile substeps (#5884)

closer towards #5522

theosanderson committed:
docs(api): Clarify that protocol.comment's message is computed in simulation (#5891)

Clarify that protocol.comment does not allow real time communication which one might otherwise guess it did. (I realise use cases where simulation might be different from reality are rare).

shlokamin committed:
feat(protocol-designer): add TC profile deck visualization (#5896)

* feat(protocol-designer): add TC profile deck visualization

add TC profile deck visualization while hovering on profile steps

#5523

* fix css linting issue

Kadee80 committed:
refactor(protocol-designer): Update confirm unsaved step deletion modals (#5877)

closes #5762

Commit:
fix(build): missing semicolon in shared-data/python/Makefile (#5901)

IanLondon committed:
feat(protocol-designer): replace v4 protocol export hint (#5897)

Closes #5888

IanLondon committed:
feat(protocol-designer): add TC announcement modal (#5898)

Closes #5887

sanni-t committed:
fix(robot-server): type mismatch in SerialCommand args (#5883)

allow Any type as args

sfoster1 committed:
chore: add mypy stuff to eslintignore (#5906)

This means we won't be trying to format internal caching stuff from
mypy, among other things

sfoster1 committed:
refactor(app): cal check: s/drop tip/drop tip in trash/ (#5894)

Commit:
feat(robot-server): Add skeleton of tip calibration session including integration test.' (#5868)

mcous committed:
refactor(*): replace black/whitelist where usage has no technical purpose (#5905)

https://tools.ietf.org/id/draft-knodel-terminology-01.html#rfc.section.1.2

ahiuchingau committed:
feat(api): define data architecture for tip length calibration in v2 (#5820)

* feat(api): define data architecture for tip length calibration in v2

closes #5606

IanLondon committed:
fix(protocol-designer): ensure all TC set temps + runProfile are awaited (#5910)

b-cooper committed:
refactor(app): add skeleton of tip length calibration view layer (#5880)

Build out the infrastructure of the new tip length calibration flow. Add views and todos, as well as
a simple test for the top level component.

sfoster1 committed:
chore(build): fix pypi test deploy again (#5913)

sfoster1 committed:
refactor(app): calcheck: confirm exit modal (#5902)

Add a modal that appears when you click on exit (in most circumstances)
during calibration check that prompts you to confirm the exit.

If the exit button is from a terminal screen, there is no confirmation.

In addition, hide the title bar exit button when it isn't necessary,
most notably when the spinner is displayed.

Note that the exit confirmation should actually lead to a generic
summary screen and not immediately exit, but that screen doesn't exist yet.

shlokamin committed:
style(protocol-designer): decrease click target area for TC form radio buttons (#5914)

* style(protocol-designer): decrease click target area for TC form radio buttons

#5885

* add back step option styling

Commit:
fix(robot-server, api): reject commands while robot is moving (#5878)

* Modification to ThreadedAsyncLock. It now runs in two modes: regular lock and forbid. Lock is normal mutex locking. Forbid will raise an error if trying to acquired the already acquired lock. This allows us to reject commands while a long running command is running eliminating deadlocks.
Also, use a Lock instead of Rlock. Rlock was not achieving locking in our async code as all calls are coming an the same thread.
* use forbid lock for /robot/home and /robot/move.
* redo use of ThreadedAsyncLock in api.
rename the decorater 'robot_is_busy' and have one implementation.
Add private CalibrationManager._move_to_front because move_to_front is called from tip_probe and we are no longer using RLock.
* remove change in fce1f794f6ee70681cb13a1368fc69a748403574. the new motion lock makes that change redundant.
* lint
* tests to prove that /robot/home and /robot/control cannot happen at the same time.

closes #5810

sfoster1 committed:
refactor(api): calcheck: 0.3mm safe height for xy checks (#5919)

This is a bit of a compromise between the current 0.5 and the user
desire for the tip to be as close to the deck as possible.

mcous committed:
fix(app-shell): clean up old robot update files (#5892)

Closes #4667

sfoster1 committed:
refactor(app): calcheck: update intro screen content (#5915)

Updates intro screen content from new designs. Tweaks the size of the
tiprack images down a bit to avoid scrollbars.

sfoster1 committed:
refactor(app): use spinnermodalpage in calcheck (#5893)

This makes the look the same as deck calibration

shlokamin committed:
fix(protocol-designer): ensure lid is closed before calling TC runProfile (#5918)

ensure lid is closed before calling TC runProfile

shlokamin committed:
style(protocol-designer): flip the expand/collapse arrow for TC substep accordions (#5925)

fix #5911

mcous committed:
refactor(app): asynchronously load config state (#5874)

Closes #5869

mcous committed:
feat(components): add position props to primitive components (#5926)

mcous committed:
refactor(app): fetch session after command in execute command epic (#5916)

b-cooper committed:
refactor(app): implement full cal check summary (#5930)

IanLondon committed:
test(labware-library): mock file-saver while using cypress (#5932)

Addresses LC side of #5457

Commit:
refactor(api,robot-server): remove aiohttp calcheck (#5907)

* remove cal check endpoints

* move SessionType to robot-server

* remove unused models

Commit:
refactor(api): remove aiohttp health and openapi (#5909)

* remove health and openapi

* remove openapi spec and validation

sfoster1 committed:
refactor(app): calcheck: update assets in check screens (#5929)

Restyles the instructions for jogging to points to make the text content side-by-side with the video, and updates the video assets to new ones containing both new content and also shrunk to work in the side-by-side layout.

Kadee80 committed:
fix(protocol-designer): Lighten TC hover overlay (#5939)

closes #5912

Kadee80 committed:
fix(protocol-designer): Update profile step fields alignment (#5937)

sfoster1 committed:
refactor(calcheck): display pipette serial numbers in summary (#5941)

Put the pipette serial in the sessions attached pipette respones. Side
note, we probably want to split these models out because some data used
by check calibration (like pipette rank) is not necessarily required by
tip length calibration.

sfoster1 committed:
refactor(app): calcheck: save report to file (#5938)

This uses the new file-saver app dependency to pop a chrome save dialog
allowing users to save the calibration check report  somewhere on disk.

In addition to the calibration check outcomes, we save the date at which
calibration check was run.

Also adds a new dependency: file-saver

This is the same package (at the same version) that we use for saving
files in PD and LL, so it shouldn't affect package locking at all.

mcous committed:
fix(app): restart MDNS browser when network interfaces change (#5933)

Fixes #5343

mcous committed:
refactor(app): do not open cal check wizard if session can't be created (#5949)

sfoster1 committed:
refactor(app): calcheck: update step outcome styling (#5948)

Displays individual step checks using similar components to the
end-of-calibration summary screen checks.

Also updates the link to the support article and displayes different
error text in cases where we can have different explicit recommendations.

ahiuchingau committed:
refactor(app): calcheck: update tip pickup animations and failure screen (#5946)

shlokamin committed:
fix(protocol-designer): auto add TC module id to TC steps (#5944)

#5942

shlokamin committed:
feat(protocol-designer): reverse multi channel substep arrows in sidebar (#5931)

b-cooper committed:
fix(api): fix transform type logic in calibration check session (#5951)

b-cooper committed:
refactor(app): always show calibration check summary on exit (#5947)

Instead of only showing the calibration check summary view upon completing the entire check, show the summary anytime a user exits the flow. Also show their progress and note any incomplete steps.

IanLondon committed:
feat(labware-library): use loadname not displayname for LC file (#5945)

Closes #5722

mcous committed:
refactor(components): remove deprecated RefreshCard in favor of hooks (#5934)

SyntaxColoring committed:
fix(app): use troubleshooting link when calcheck fails at tip pickup (#5952)

IanLondon committed:
feat(labware-library): use slot 2 instead of 3 in LC test protocol (#5950)

Closes #5019

ahiuchingau committed:
refactor(app): calcheck: add tooltip over table header (#5953)

Laura-Danielle committed:
feat(api, app): Check Robot Deck Transform (#5845)

Laura-Danielle committed:
fix(api): Simulate protocols with a realistic deck transform (#5908)

mcous committed:
refactor(app): move cal session constants and types to sessions (#5956)

Laura-Danielle committed:
fix(api): Fix failing test from edge due to kwarg check (#5961)

b-cooper committed:
refactor(app): add troubleshooting instructions to cal check summary (#5959)

Co-authored-by: Seth Foster <seth@opentrons.com>

mcous committed:
refactor(app): wire deck cal status API to UI  (#5962)

This PR wires GET /calibration/status to the robot controls card and associated UI.


Note: Tooltips are not working because out Button components still can't display a tooltip if they're disabled, because disabling them sets pointer-events: none in CSS. I will try to follow up and I have written this UI + tests to show tooltips as soon as that is fixed. Give the code a look though for that tooltip copy.

Co-authored-by: Seth Foster <seth@opentrons.com>

b-cooper committed:
refactor(app): add calibration and instrument data to cal check report (#5964)

Add more helpful information to the final JSON report that users can download at the end of the Robot Calibration Check flow.

Co-authored-by: Mike Cousins <mike@opentrons.com>
Co-authored-by: Mike Cousins <mike@cousins.io>

b-cooper committed:
feat(app): introduce Robot Calibration Check tool (#5960)

Adds a wizard flow to the robot page that walks a user through diagnosing the root cause of
movement-related issues they may have encountered.


Co-authored-by: Seth Foster <seth@opentrons.com>

sfoster1 committed:
chore(bump): 3.19.0-alpha.2 (#5965)

shlokamin reviewed (approved):
LGTM
PD changelog is being updated for things that haven't been released yet, but this is a known issue.

mcous committed:
refactor(app): fix race condition introduced by async anayltics init (#5971)

mcous committed:
chore(release): 3.19.0-alpha.3 (#5973)

sfoster1 committed:
chore(release): 3.19.0-alpha.4 (#5977)

sfoster1 committed:
fix(app): calcheck: content fixups (#5989)

- Correctly describe the calibration check feature on the robot controls
page
- Correctly recommend deck calibration if tip pickup is incorrect

sfoster1 committed:
fix(app): calcheck: fix display of bad deck transform (#5988)

Looks like 2bb54a999d682da246884c87bcf29075058878a3 accidentally changed
the value for a bad deck transform value from the api, might have been
an overzealous find and replace. This value should match the value in
opentrons.calibration.helper_classes

b-cooper committed:
fix(app): replace wrong assets for xy multi cal check (#5996)

ahiuchingau committed:
feat(api): add warning when a wrong tiprack is used for the pipette in the protocol (#5997)

* feat(api): add warning when a wrong tiprack is used for the pipette in the protocol

* add check protocol to LabwareHeightError

* use tiprack vol to check pip & tiprack combo

sfoster1 committed:
chore(bump): v3.19.0-alpha.5 (#6000)

SyntaxColoring committed:
docs(api): correct .move() example (#6002)

Fixes #5806.

ethanfjones-zz committed:
Reordered navigation sidebar (#6009)

Reordered the navigation sidebar to reflect how we suggest a protocol should be organized in the How it's Organized section

b-cooper committed:
fix(app): fix clunky text wrapping for multi tip pick up in cal check (#6026)

sfoster1 committed:
chore(release): v3.19.0 (#6027)

nusrat813 reviewed (approved):
LGTM

bndo reviewed (approved):
🥇 lgtm
